File: pkg/analyzers/couples/tc.go
package couples

// RenamePair represents a file rename detected in a single commit.
type RenamePair struct {
	FromName string
	ToName   string
}

// CommitData is the per-commit TC payload emitted by Consume().
// It captures the coupling context, author-file touches, renames,
// and whether the author's commit count was incremented.
type CommitData struct {
	// CouplingFiles is the list of files forming the coupling context
	// (already filtered by CouplesMaximumMeaningfulContextSize).
	CouplingFiles []string
	// AuthorFiles maps file name to touch count for this commit's author.
	AuthorFiles map[string]int
	// Renames holds rename pairs detected in this commit.
	Renames []RenamePair
	// CommitCounted is true when this commit incremented the author's commit count.
	CommitCounted bool
}

// TickData is the per-tick aggregated payload stored in analyze.TICK.Data.
type TickData struct {
	// Files maps file -> otherFile -> co-occurrence count.
	Files map[string]map[string]int
	// People is per-author file touch counts, indexed by author ID.
	People []map[string]int
	// PeopleCommits is per-author commit counts, indexed by author ID.
	PeopleCommits []int
	// Renames accumulated during this tick.
	Renames []RenamePair
}

File: crates/cf-couples/src/tc.rs
//! Per-commit and per-tick transfer types.

use std::collections::BTreeMap;

/// A file rename detected in a single commit.
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct RenamePair {
    /// Original path.
    pub from_name: String,
    /// New path.
    pub to_name: String,
}

/// Per-commit summary used for timeseries output.
///
/// JSON field names: `files_touched`, `author_id`.
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct CommitSummary {
    /// Number of files in the coupling context.
    pub files_touched: usize,
    /// Author index.
    pub author_id: usize,
}

/// Per-commit transfer payload emitted by the pipeline consume step.
#[derive(Debug, Clone, Default, PartialEq, Eq)]
pub struct CommitData {
    /// Files forming the coupling context (already context-size filtered).
    pub coupling_files: Vec<String>,
    /// File name → touch count for this commit's author.
    ///
    /// A `BTreeMap` is used for deterministic iteration; only `1` is ever
    /// stored per file and downstream accumulation is additive, so ordering
    /// does not affect results.
    pub author_files: BTreeMap<String, i64>,
    /// Rename pairs detected in this commit.
    pub renames: Vec<RenamePair>,
    /// Whether this commit incremented the author's commit count.
    pub commit_counted: bool,
}

/// Per-tick aggregated payload.
#[derive(Debug, Clone, Default)]
pub struct TickData {
    /// `file -> otherFile -> co-occurrence count`.
    pub files: BTreeMap<String, BTreeMap<String, i64>>,
    /// Per-author file touch counts, indexed by author ID.
    pub people: Vec<BTreeMap<String, i64>>,
    /// Per-author commit counts, indexed by author ID.
    pub people_commits: Vec<i64>,
    /// Renames accumulated during this tick.
    pub renames: Vec<RenamePair>,
    /// Per-commit summaries keyed by commit hash string.
    pub commit_stats: BTreeMap<String, CommitSummary>,
}
